benchmarking/scripts/benchmark_plot.py

```diff
@@ -0,0 +1,89 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import argparse
+import json
+
+from pathlib import Path
+from io import StringIO
+
+
+def timer_file_to_dataframe(timer_file_path):
+    """This method read in a LFRic timer file into a pandas data frame"""
+
+    with open(timer_file_path, "r") as file:
+        data = file.read()
+        data = data.replace("||", ",")
+        data = data.replace("=", "")
+    df = pd.read_csv(StringIO(data), sep=",", header=0)
+    df = df.iloc[:, 1:-1]
+    headings_dict = {}
+    for heading in df.columns:
+        headings_dict[heading] = heading.strip()
+    df.rename(columns=headings_dict, inplace=True)
+
+    return df
+
+
+def generate_most_expensive_cost_plot(timer_file_path, output_file_name, plot_label):
+    """This method generates a plot of the top 15 most costly routine
+    calls as a histogram plot."""
+
+    # Process the LFRic timer file into a pandas data frame
+    df = timer_file_to_dataframe(timer_file_path=timer_file_path)
+
+    # Sort the dataframe in descending order of expensive routine calls
+    df.sort_values(by=["mean time(s)"], inplace=True, ascending=False)
+
+    # Create a bar chart of the 15 most expensive routine calls
+    routine = df["Routine"].head(15)
```

Comment:
I'm being picky but it would be ideal to have this number as an additional argument to the script (and set it to 15 by default)